File: app/src/main/java/org/tasks/etesync/CreateCalendarViewModel.java
package org.tasks.etesync;

import org.tasks.data.CaldavAccount;
import org.tasks.ui.CompletableViewModel;

@SuppressWarnings("WeakerAccess")
public class CreateCalendarViewModel extends CompletableViewModel<String> {
  void createCalendar(EteSyncClient client, CaldavAccount account, String name) {
    run(() -> client.forAccount(account).makeCollection(name));
  }
}

File: app/src/main/java/org/tasks/etesync/CreateCalendarViewModel.kt
package org.tasks.etesync

import dagger.hilt.android.lifecycle.HiltViewModel
import org.tasks.data.CaldavAccount
import org.tasks.ui.CompletableViewModel
import javax.inject.Inject

@Deprecated("use etebase")
@HiltViewModel
class CreateCalendarViewModel @Inject constructor(
        private val client: EteSyncClient) : CompletableViewModel<String?>() {
    suspend fun createCalendar(account: CaldavAccount, name: String, color: Int) {
        run { client.forAccount(account).makeCollection(name, color) }
    }
}